presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -402,17 +411,17 @@ public PlanOptimizers(
                                 new TransformUncorrelatedInPredicateSubqueryToDistinctInnerJoin(),
                                 new TransformUncorrelatedInPredicateSubqueryToSemiJoin(),
                                 new TransformCorrelatedScalarAggregationToJoin(metadata.getFunctionAndTypeManager()),
-                                new TransformCorrelatedLateralJoinToJoin())),
+                                new TransformCorrelatedLateralJoinToJoin(metadata.getFunctionAndTypeManager()))),
```

Comment:
metadata.getFunctionAndTypeManager() is being called multiple times, perhaps store it in a variable

Reply:
It'll change too much, lets do this in a separate PR after this